rebuild/src/secret_key.cc

```diff
@@ -0,0 +1,171 @@
+            "fromKeygen",
+            &SecretKey::FromKeygen,
+            static_cast<napi_property_attributes>(
+                napi_static | napi_enumerable)),
+        StaticMethod(
+            "deserialize",
+            &SecretKey::Deserialize,
+            static_cast<napi_property_attributes>(
+                napi_static | napi_enumerable)),
+        InstanceMethod(
+            "serialize",
+            &SecretKey::Serialize,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "toPublicKey",
+            &SecretKey::ToPublicKey,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "sign",
+            &SecretKey::Sign,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+    };
+
+    Napi::Function ctr = DefineClass(env, "SecretKey", proto, module);
+    module->_secret_key_ctr = Napi::Persistent(ctr);
+    exports.Set(Napi::String::New(env, "SecretKey"), ctr);
+
+    exports.Get("BLST_CONSTANTS")
+        .As<Napi::Object>()
+        .Set(
+            Napi::String::New(env, "SECRET_KEY_LENGTH"),
+            Napi::Number::New(env, BLST_TS_SECRET_KEY_LENGTH));
+}
+
+Napi::Value SecretKey::FromKeygen(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE(info, env, module)
+    Napi::Value ikm_value = info[0];
+
+    BLST_TS_UNWRAP_UINT_8_ARRAY(ikm_value, ikm, "ikm")
+    // Check for less than 32 bytes so consumers don't accidentally create
+    // zero keys 
+    if (ikm.ByteLength() < BLST_TS_SECRET_KEY_LENGTH) {
+        std::ostringstream msg;
+        msg << "ikm must be greater than or equal to "
+            << BLST_TS_SECRET_KEY_LENGTH << " bytes";
+        Napi::TypeError::New(env, msg.str()).ThrowAsJavaScriptException();
+        return env.Undefined();
+    }
+
+    bool is_external = true;
+    Napi::Object wrapped = module->_secret_key_ctr.New(
+        {Napi::External<bool>::New(env, &is_external)});
+    SecretKey *sk = SecretKey::Unwrap(wrapped);
+
+    // If `info` string is passed use it otherwise use default without
+    if (!info[1].IsUndefined()) {
+        if (!info[1].IsString()) {
+            Napi::TypeError::New(env, "BLST_ERROR: info must be a string")
+                .ThrowAsJavaScriptException();
+            return env.Undefined();
+        }
+        sk->_key->keygen(
+            ikm.Data(),
+            ikm.ByteLength(),
+            info[1].As<Napi::String>().Utf8Value());
+    } else {
+        sk->_key->keygen(ikm.Data(), ikm.ByteLength());
+    }
+
+    // Check if key is zero and set flag if so. Several specs depend on this
+    // check (signing with zero key). Do after building instead of checking
+    // incoming bytes incase info changes the key
+    blst::byte key_bytes[BLST_TS_SECRET_KEY_LENGTH];
+    sk->_key->to_bendian(key_bytes);
+    if (is_zero_bytes(key_bytes, 0, BLST_TS_SECRET_KEY_LENGTH)) {
+        sk->_is_zero_key = true;
+    }
```

Comment:
```suggestion
    sk->_is_zero_key = is_zero_bytes(key_bytes, 0, BLST_TS_SECRET_KEY_LENGTH);
```

Reply:
updated in 2e41be9123e8d450bbe04e269d340008f10c876c